src/rust/ensogl/lib/select/src/entry.rs

```diff
@@ -0,0 +1,310 @@
+    #[derivative(Debug="ignore")]
+    pub icon        : Option<display::object::Any>,
+}
+
+impl Model {
+    /// Create model of simple entry with given label.
+    ///
+    /// The model won't have icon nor higlighting, but those can be set using `highlight` and
+    /// `with_icon`.
+    pub fn new(label:impl Str) -> Self {
+        Self {
+            label       : label.into(),
+            highlighted : default(),
+            icon        : default(),
+        }
+    }
+
+    /// Add highlighting to the entry and return it.
+    pub fn highlight(mut self, bytes:impl IntoIterator<Item=TextRange<Bytes>>) -> Self {
+        self.highlighted.extend(bytes.into_iter());
+        self
+    }
+
+    /// Add icon to the entry and return it.
+    pub fn with_icon(mut self, icon:impl display::Object + 'static) -> Self {
+        self.icon = Some(icon.into_any());
+        self
+    }
+}
+
+
+// === Entry Model Provider ===
+
+/// The Entry Model Provider for select component.
+///
+/// The select does not display all entries at once, instead it lazily ask for models of entries
+/// when they're about to be displayed. So setting the select content is essentially providing
+/// implementor of this trait.
+pub trait ModelProvider : Debug {
+    /// Number of all entries.
+    fn entry_count(&self) -> usize;
+
+    /// Get the model of entry with given id.
+    fn get(&self, id:Id) -> Model;
+}
+
+/// A wrapper for shared instance of some ModelProvider.
+#[derive(Clone,CloneRef,Debug,Shrinkwrap)]
+pub struct AnyModelProvider(Rc<dyn ModelProvider>);
+
+impl<T:ModelProvider + 'static> From<T> for AnyModelProvider {
+    fn from(provider: T) -> Self { Self(Rc::new(provider)) }
+}
+
+impl Default for AnyModelProvider {
+    fn default() -> Self {
+        let logger = Logger::new("EmptyModelProvider");
+        EmptyProvider{logger}.into()
+    }
+}
+
+
+// === Empty Model Provider ===
+
+/// An Entry Model Provider giving no entries.
+///
+/// This is the default provider for new select components.
+#[derive(Clone,CloneRef,Debug)]
+pub struct EmptyProvider {
+    logger : Logger,
+}
+
+impl ModelProvider for EmptyProvider {
+    fn entry_count(&self) -> usize { 0 }
+    fn get(&self, id:usize) -> Model {
+        error!(self.logger, "Getting {id} from empty provider!");
+        Model {
+            label       : "Invalid".to_string(),
+            highlighted : default(),
+            icon        : None,
+        }
+    }
+}
+
+
+
+// =============
+// === Entry ===
+// =============
+
+/// A displayed entry in select component.
+///
+/// The Display Object position of this component is docked to the middle of left entry's boundary.
+/// It differs from usual behaviour of EnsoGl components, but makes the entries alignment much
+/// simpler.
+#[derive(Clone,CloneRef,Derivative)]
+#[derivative(Debug)]
+pub struct Entry {
+    id             : Rc<Cell<Option<Id>>>,
+    label          : text::Area,
+    #[derivative(Debug="ignore")]
```

Comment:
why? This should be fine without this line. If not, it means we are missing debug impl somewhere.